From 1a33c3756e2796ddfda7392a78aaa87523e6f575 Mon Sep 17 00:00:00 2001
From: Feng Zhang <feng@radxa.com>
Date: Mon, 2 Mar 2026 11:04:23 +0000
Subject: [PATCH 14/14] board: sunxi: cubie_board: Initialize GPIO state to
 input

Signed-off-by: Feng Zhang <feng@radxa.com>
---
 src/board/sunxi/cubie_board.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/board/sunxi/cubie_board.c b/src/board/sunxi/cubie_board.c
index 61bb55af97e..29e9868088e 100644
--- a/src/board/sunxi/cubie_board.c
+++ b/src/board/sunxi/cubie_board.c
@@ -1,4 +1,5 @@
 #include <env.h>
+#include <sunxi_gpio.h>
 #include <asm-generic/gpio.h>
 #include <sunxi_gpadc.h>
 #include <dm/device.h>
@@ -42,6 +43,8 @@ void radxa_set_board_type(void)
 {
 	int i, vol, level;
 
+	sunxi_gpio_set_cfgpin(SUNXI_GPK(24), SUNXI_GPIO_INPUT);
+
 	if (gpio_request(HW_ID_GPIO, "hw_id_level") != 0) {
 		debug("Failed to request GPIO %d for hardware ID detection\n", HW_ID_GPIO);
 		return;
-- 
2.49.0

