From 3b55811042c6c243a3b90610c3e0b238c2d6cc8a Mon Sep 17 00:00:00 2001
From: Feng Zhang <feng@radxa.com>
Date: Wed, 4 Mar 2026 09:44:33 +0000
Subject: [PATCH 2/2] drivers/misc: radxa-i2c-eeprom: Fix macaddr read error

Signed-off-by: Feng Zhang <feng@radxa.com>
---
 drivers/misc/radxa-i2c-eeprom.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/misc/radxa-i2c-eeprom.c b/src/drivers/misc/radxa-i2c-eeprom.c
index 841641ca443..766003f69bd 100644
--- a/src/drivers/misc/radxa-i2c-eeprom.c
+++ b/src/drivers/misc/radxa-i2c-eeprom.c
@@ -487,7 +487,7 @@ int radxa_mac_read_from_eeprom(u8 *macaddr)
 
 	if(read_eeprom(eeprom_buff))
 		return -1;
-    memcpy(macaddr, einfo_page2.mac_addr, strlen(einfo_page2.mac_addr) + 1);
+    memcpy(macaddr, einfo_page2.mac_addr, 6);
 	return 0;
 }
 
-- 
2.49.0

